src/charm.py

```diff
@@ -1558,12 +1581,56 @@ def is_relation_feasible(self, rel_interface) -> bool:
             )
             return False
 
+        if self.get_cluster_mismatched_revision_status():
+            self.unit.status = self.get_cluster_mismatched_revision_status()
+            return False
+
         return True
 
     def is_sharding_component(self) -> bool:
         """Returns true if charm is running as a sharded component."""
         return self.is_role(Config.Role.SHARD) or self.is_role(Config.Role.CONFIG_SERVER)
 
+    def get_cluster_mismatched_revision_status(self) -> Optional[StatusBase]:
+        """Returns a Status if the cluster has mismatched revisions."""
+        if self.version_checker.is_local_charm(self.app.name):
+            logger.warning(LOCALLY_BUIT_CHARM_WARNING)
+            return
+
+        if self.version_checker.is_integrated_to_locally_built_charm():
+            logger.warning(INTEGRATED_TO_LOCALLY_BUIT_CHARM_WARNING)
+            return
+
+        try:
+            # check for invalid versions in sharding integrations, i.e. a shard running on
+            # revision  88 and a config-server running on revision 110
+            this_charms_version = get_charm_revision(self.unit)
```

Comment:
I believe we can take it out of the try / except block + nit: renamed variable
```suggestion
        # check for invalid versions in sharding integrations, i.e. a shard running on
        # revision  88 and a config-server running on revision 110
        current_charm_revision = get_charm_revision(self.unit)
        try:
```